curation_concerns-models/app/services/curation_concerns/persist_directly_contained_output_file_service.rb

```diff
@@ -0,0 +1,28 @@
+module CurationConcerns
+  # This Service is an implementation of the Hydra::Derivatives::PeristOutputFileService
+  # It supports directly contained files
+  class PersistDirectlyContainedOutputFileService < Hydra::Derivatives::PersistBasicContainedOutputFileService
+    # This method conforms to the signature of the .call method on Hydra::Derivatives::PeristOutputFileService
+    # * Persists the file within the object at destination_name
+    #
+    # NOTE: Uses basic containment. If you want to use direct containment (ie. with PCDM) you must use a different service (ie. Hydra::Works::AddFileToGenericFile Service)
+    #
+    # @param [#read] stream the data to be persisted
+    # @param [Hash] directives directions which can be used to determine where to persist to.
+    # @option directives [String] url URI for the parent object.
+    # @option directives [String] container Name of the container association.
+    def self.call(stream, directives)
```

Comment:
This method seems to be doing a lot. I wouldn't hold up the PR because of this, but might there be some stuff you could break out into a private method?